### ISSUE TITLE
Start of the rains in Version 4.18

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4689]
I really like the facilities now - for the 3 columns being possible as output.

I tried with Dodoma, starting from 1 November to 31 December, with 20mm in 3 days.

There are some minor issues still  - I am afraid!
1. I would like the option of displaying the logical column.  
2. Does the control have to be quite so wide?  So could all 3 come on the same line, each with its checkbox.  So something like
Display: [tick-box] Day of Year [box]  [tick-box] Date [box] [tick-box] Occurrence [box] 
3.  The years are now not in ascending order.  The years of no occurrence are all at the bottom.  Easy to sort and that seems to give no problems with other summaries.
4. The date is not adjusting for missing values.  In the example above 2012 has November missing, so the doy is correctly given as NA.  But the date is given as 9 Dec - also correct if the missing values are ignored - but they should be considered.

**Comment by dannyparsons** [comment]
3. is what will be fixed by the next version of `dplyr` when it is released.

**Comment by shadrackkibet** [comment]
Here is the current design:

![image](https://user-images.githubusercontent.com/26301973/41587581-353becd2-73b8-11e8-9419-6a763e4e15e5.png)

**Comment by shadrackkibet** [comment]
@rdstern any comment on this design?

**Comment by rdstern** [comment]
Yes, I liked the 3 columns for saving going across the dialogue.  I wonder if the receivers need to be quite so wide, but I guess that's a general size?  It might make the end dialogue wider than it is now.  There we have the end of rains and end of season.

**Comment by shadrackkibet** [comment]
The size of input textboxes is not general i will adjust that for end of rains so that the dialog is not wide.

**Comment by rdstern** [comment]
With the Dodoma data I tried **saving all 3 columns**.  The period is from 1 October to end December 

If I do the start with 20mm in 3 days and also max dry spell of 7 in 21 days it works OK.

But I first do the start without the dry spells condition.  That worked fine. 
Then I returned to the dialogue and added the dry spell condition as above.  I also changed the names by adding 1 to each name - so it doesn't overwrite.   Now it gives an error.  It then produces the first 2 columns, but not the logical status column.

**Comment by rdstern** [comment]
Great.  Many thanks.
